db_checker.py

```diff
@@ -286,41 +276,30 @@ def test_log_info_tags(self):
                         if info_name.startswith("log"):
                             previous_source = log_fields.get(info_name, None)
                             if previous_source is not None:
-                                print "ERROR: Invalid logging config: {source} repeats the log info tag " \
-                                      "{tag} (originally in {orig})".format(source=err_src_fmt(db, rec), tag=info_name,
-                                                                            orig=err_src_fmt(*previous_source))
+                                failure_message += "   -> Invalid logging config: {source} repeats the log info tag " \
+                                      "{tag}\n".format(source=rec, tag=info_name)
                                 err += 1
                             else:
                                 log_fields[info_name] = (db, rec)
+
                         if info_name == "log_period_seconds" or info_name == "log_period_pv":
                             if logging_period is None:
                                 logging_period = (db, rec)
                             else:
-                                print "ERROR: Invalid logging config: {source} alters the logging period type " \
-                                    "(originally in {orig})".format(source=err_src_fmt(db, rec),
-                                                                    tag=info_name, orig=err_src_fmt(*logging_period))
+                                failure_message += "   -> Invalid logging config: {source} alters the logging period " \
+                                                   "type\n".format(source=rec, tag=info_name)
                                 err += 1
 
-        self.assertEqual(err, 0, msg="LOG infos repeated")
+        self.assertEqual(err, 0, msg=failure_message)
 
 
 def set_up(directories):
     """
-    This set up method loads and parses the db and template files prior to testing.
+    This set up method generates parsed DB and template files.
     """
-
-    global dbs
-
-    dbs = FileHolder()
-
     for directory in directories:
-        for file_type in ['.db', '.template']:
-            dbs.load_files(directory, file_type)
-
-    # dbs.saveChecked()
-    dbs = dbs.parse_files()
-
-    print "Number of EPICS dbs Found: " + str(len(dbs))
+        for parsed_file in parsed_files(directory, ['.db', '.template']):
```

Comment:
parsed_files can raise a `ValueError` which doesn't appear to be explicitly caught. Is that intentional?

Reply:
The only case where you'd get a ValueError is when you have a serious problem with a DB file (e.g. truncation). In which case, you would want to see the full stack trace so that you can see where things went wrong, and the python shell to exit with a failure code. 

Of course, I could print the traceback and exit failure manually, but that's exactly what the python interpreter does that by default with an unhandled exception anyway so I didn't see what is gained by doing it manually.